smarts/core/agents_provider.py

```diff
@@ -122,19 +122,29 @@ def perform_agent_actions(self, agent_actions: Dict[str, Any]):
         Args:
             agent_actions: a dictionary from each agent_id to its actions for this step.
         """
-        self._remove_missing_actors(agent_actions)
-        actions_without_agents = agent_actions.keys() - self._my_agent_actors.keys()
-        assert (
-            not actions_without_agents
-        ), f"actions specified for non-tracked agents:  {actions_without_agents}"
+        self._remove_actors_without_actions(agent_actions)
+        agents_without_actors = agent_actions.keys() - self._my_agent_actors.keys()
+        if agents_without_actors:
+            if self._log.level >= logging.ERROR:
```

Comment:
Why do we need to check the log level? Won't it be filtered out already based on whatever level is set?

Reply:
Yes, thanks for pointing it out. I meant to only lock the error warning behind the logging level.